e2e/create-cluster_test.go

```diff
@@ -226,6 +238,58 @@ var _ = Describe("CreateCluster", func() {
 		}, 5*time.Minute, 5*time.Second).Should(Succeed(), "waiting for expected readiness.")
 	}
 
+	waitForTenantAccess := func(numExpectedNodes int) {
```

Comment:
Can we break this into two functions. One that returns a working kubeconfig, and another one that can use the guest kubeconfig to verify the number of nodes?

My reasoning here is that we'll want to do more actions with this guest kubeconfig in the future, so we might as well structure the code in a way that allows us to gain access and pass around the guest kubeconfig now.